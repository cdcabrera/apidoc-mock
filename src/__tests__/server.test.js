const server = require('../server');

describe('logger', () => {
  it('should return specific properties', () => {
    expect(server).toMatchSnapshot('specific properties');
  });

  it('should setHeaders', () => {
    const mockFunction = jest.fn();
    const mockResponse = { end: mockFunction, set: mockFunction };

    server.setHeaders({ headers: {} }, mockResponse, mockFunction);
    expect(mockFunction.mock.calls).toMatchSnapshot('default headers');
    mockFunction.mockClear();

    server.setHeaders({ headers: { origin: 'dolor sit' } }, mockResponse, mockFunction);
    expect(mockFunction.mock.calls).toMatchSnapshot('origin headers');
    mockFunction.mockClear();

    server.setHeaders({ headers: { 'access-control-request-headers': 'lorem ipsum' } }, mockResponse, mockFunction);
    expect(mockFunction.mock.calls).toMatchSnapshot('allow headers');
    mockFunction.mockClear();

    server.setHeaders({ method: 'OPTIONS' }, mockResponse, mockFunction);
    expect(mockFunction.mock.calls).toMatchSnapshot('request method');
    mockFunction.mockClear();
  });

  it('should setResponse', () => {
    const mockLogger = jest.fn();
    const defaultOutput = server.setResponses(undefined, undefined, { logger: { info: mockLogger } });
    expect({
      output: defaultOutput,
      mock: mockLogger.mock.calls
    }).toMatchSnapshot('default');

    const mockServer = jest.fn();
    const basicServer = server.setResponses(
      { get: mockServer, listen: mockServer, on: mockServer },
      { port: 3000 },
      {
        buildResponses: () => [{ type: 'get', url: '/lorem/ipsum', callback: jest.fn() }]
      }
    );
    expect({
      output: basicServer,
      mock: mockServer.mock.calls
    }).toMatchSnapshot('basicServer');
  });

  it('should setServer', async () => {
    const mockFunction = jest.fn();
    await expect(async () =>
      server.setServer(
        { port: 8000, watch: ['./testFile.js'] },
        { logger: { error: mockFunction }, setResponses: mockFunction }
      )
    ).rejects.toThrow('Server failed to load');

    expect({
      mock: mockFunction.mock.calls
    }).toMatchSnapshot('default');
  });
});
