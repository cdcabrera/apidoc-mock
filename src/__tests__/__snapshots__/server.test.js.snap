// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger should return specific properties: specific properties 1`] = `
{
  "setHeaders": [Function],
  "setResponses": [Function],
  "setServer": [Function],
}
`;

exports[`logger should setHeaders: allow headers 1`] = `
[
  [
    "Access-Control-Allow-Origin",
    "*",
  ],
  [
    "Access-Control-Allow-Credentials",
    true,
  ],
  [
    "Access-Control-Allow-Methods",
    "GET, PUT, POST, DELETE, HEAD, OPTIONS, PATCH",
  ],
  [
    "Access-Control-Allow-Headers",
    "lorem ipsum",
  ],
  [],
]
`;

exports[`logger should setHeaders: default headers 1`] = `
[
  [
    "Access-Control-Allow-Origin",
    "*",
  ],
  [
    "Access-Control-Allow-Credentials",
    true,
  ],
  [
    "Access-Control-Allow-Methods",
    "GET, PUT, POST, DELETE, HEAD, OPTIONS, PATCH",
  ],
  [],
]
`;

exports[`logger should setHeaders: origin headers 1`] = `
[
  [
    "Access-Control-Allow-Origin",
    "dolor sit",
  ],
  [
    "Access-Control-Allow-Credentials",
    false,
  ],
  [
    "Access-Control-Allow-Methods",
    "GET, PUT, POST, DELETE, HEAD, OPTIONS, PATCH",
  ],
  [],
]
`;

exports[`logger should setHeaders: request method 1`] = `
[
  [
    "Access-Control-Allow-Origin",
    "*",
  ],
  [
    "Access-Control-Allow-Credentials",
    true,
  ],
  [
    "Access-Control-Allow-Methods",
    "GET, PUT, POST, DELETE, HEAD, OPTIONS, PATCH",
  ],
  [],
]
`;

exports[`logger should setResponse: basicServer 1`] = `
{
  "mock": [
    [
      "/lorem/ipsum",
      [MockFunction],
    ],
    [
      3000,
      [Function],
    ],
  ],
  "output": undefined,
}
`;

exports[`logger should setResponse: default 1`] = `
{
  "mock": [
    [
      "waiting",
    ],
  ],
  "output": undefined,
}
`;

exports[`logger should setServer: default 1`] = `
{
  "mock": [
    [
      [Function],
    ],
    [
      "Error, confirm settings:
port=8000
watch=./testFile.js",
    ],
  ],
}
`;
